app/src/components/messageParser.js

```diff
@@ -15,10 +22,26 @@ const grokParse = (pattern, message) => {
 };
 
 const messageParser = {
-  /** Parse an incoming log request into HALPAS JSON object
-     *  @param {object} obj A JSON Object, pre-validated to contain expected fields
-     *  @returns {object} Returns a JSON Object for HALPAS logstash
-     */
+  /**
+   *  @function parseMany
+   *  Parse an array of incoming log requests into an array HALPAS JSON object
+   *  @param {string} authorizedParty A string representing the authorized party
+   *  @param {object[]} obj A JSON Array, pre-validated to contain expected fields
+   *  @returns {object[]} Returns an array of JSON Object for HALPAS logstash
+   */
+  parseMany: async (authorizedParty, obj) => {
+    return await Promise.all(
```

Comment:
I don't think we want to do a promise all here, we create a timestamp for when get called, the assumption being the caller is sending in data in order.  Timestamps can get out of sync (milliseconds), but data could be represented 'out of order'.

Reply:
We can make that synchronous - and as per discussion down the road consider implementing a log order field or similar. Will look into it on next PR.